core/src/main/groovy/com/novoda/gradle/release/ReleasePlugin.groovy

```diff
@@ -4,20 +4,48 @@ import com.jfrog.bintray.gradle.BintrayPlugin
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.publish.maven.MavenPublication
-import org.gradle.util.GradleVersion
 
 class ReleasePlugin implements Plugin<Project> {
 
+    @Override
     void apply(Project project) {
         PublishExtension extension = project.extensions.create('publish', PublishExtension)
         project.afterEvaluate {
+            checkClosureSetup(extension)
             project.apply([plugin: 'maven-publish'])
             attachArtifacts(extension, project)
             new BintrayPlugin().apply(project)
             new BintrayConfiguration(extension).configure(project)
         }
     }
 
+    /**
+     * Give the user quicker and more obvious feedback when they
+     * haven't set their project up correctly
+     */
+    private static void checkClosureSetup(PublishExtension extension) {
+        String extensionError = "";
+        if (extension.userOrg == null) {
+            extensionError += "Missing userOrg. "
+        }
+        if (extension.groupId == null) {
+            extensionError += "Missing groupId. "
+        }
+        if (extension.artifactId == null) {
+            extensionError += "Missing artifactId. "
+        }
+        if (extension.publishVersion == null) {
+            extensionError += "Missing publishVersion. "
+        }
+        if (extension.desc == null) {
+            extensionError += "Missing desc. "
+        }
+        if (extensionError) {
+            String prefix = "Have you created the publish closure? "
+            throw new IllegalStateException(prefix + extensionError)
```

Comment:
Are they all mandatory? I was thinking the description can be optional.

Reply:
yes all mandatory

description is mandatory https://github.com/novoda/bintray-release/blob/master/core/src/main/groovy/com/novoda/gradle/release/PublishExtension.groovy#L31